GitCommands/RevisionReader.cs

```diff
@@ -448,32 +445,40 @@ DateTime ParseUnixDateTime()
             return true;
         }
 
-        private static (string? body, string? additionalData) ParseCommitBody(StringLineReader reader, string subject)
+        private static (string? subject, string? body, string? additionalData) ParseCommitBody(StringLineReader reader)
         {
-            int lengthOfSubjectRepeatedInBody = subject.Length + 2/*newlines*/;
-            if (reader.Remaining == lengthOfSubjectRepeatedInBody + EndOfBody.Length)
+            if (reader.Remaining == 0)
             {
-                return (body: subject, additionalData: null);
+                // TODO log this error
+                Debug.Fail("Subject is empty in the log -- this should not happen");
+                return (subject: null, body: null, additionalData: null);
             }
 
-            string tail = reader.ReadToEnd() ?? "";
-            int indexOfEndOfBody = tail.LastIndexOf(EndOfBody, StringComparison.InvariantCulture);
+            // Ignore leading empty or whitespace lines, see Git pretty.c
+            string tail = reader.ReadToEnd().TrimStart() ?? "";
```

Comment:
Is there a way to prevent allocating this tail string? Perhaps by extending the reader? For large repositories, this may create a lot of objects for the GC.

Eg:

reader.SkipWhiteSpace();
var body = reader.ReadTo(EndOfBody);

Reply:
For end of body, I would like to remove that with a single null, and read additional data separately. Not finished yet. 

This implementation should have the same allocations as the current.

Reply:
Changing to something like the suggested to simplify further, will simplify updated parsing of the reader. 

ParseCommitBody() is no longer needed, removing it and the tests for it. 
This may be be replaced with parsing the git-log output instead eventually.

Reply:
Nice.